cloudformation.yaml

```diff
@@ -84,37 +71,26 @@ Resources:
                 - autoscaling:AttachInstances
                 - autoscaling:TerminateInstanceInAutoScalingGroup
                 Resource:
-                  - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/${DataNodeAsg}
-              - !If
-                  - HasDedicatedMasters
-                  -
-                    Effect: Allow
-                    Action:
-                    - autoscaling:DetachInstances
-                    - autoscaling:AttachInstances
-                    - autoscaling:TerminateInstanceInAutoScalingGroup
-                    Resource:
-                    - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/${DedicatedMasterAsg}
-                  - !Ref AWS::NoValue
-              - !If
-                  - HasWarmDataNodes
-                  -
-                    Effect: Allow
-                    Action:
-                    - autoscaling:DetachInstances
-                    - autoscaling:AttachInstances
-                    - autoscaling:TerminateInstanceInAutoScalingGroup
-                    Resource:
-                    - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/${WarmDataNodeAsg}
-                  - !Ref AWS::NoValue
+                - !Sub arn:aws:autoscaling:${AWS::Region}:${AWS::AccountId}:autoScalingGroup:*:autoScalingGroupName/*
+                Condition:
+                  StringLike:
+                    autoscaling:ResourceTag/RotateWithElasticsearchNodeRotation: "*"
```

Comment:
I like the idea of using this tag to continue restricting the permissions to the relevant ASGs only 👍